py/desispec/workflow/os.py

```diff
@@ -0,0 +1,117 @@
+
+
+import os
+import numpy as np
+import glob
+import json
+import psutil
+
+
+pathjoin = os.path.join
+
+def listpath(*args):
+    # return np.sort(os.listdir(pathjoin(*args))).tolist()
+    if os.path.exists(pathjoin(*args)):
+        srtlist = sorted(os.listdir(pathjoin(*args)))
+        if '.DS_Store' in srtlist:
+            srtlist.remove('.DS_Store')
+        return srtlist
+    else:
+        return []
+
+
+def globpath(*args):
+    # return np.sort(glob.glob(pathjoin(*args))).tolist()
+    if os.path.exists(pathjoin(*args)):
+        srtlist = sorted(glob.glob(pathjoin(*args)))
+        if '.DS_Store' in srtlist:
+            srtlist.remove('.DS_Store')
+        return srtlist
+    else:
+        return []
+
+
+def get_file_list(filename, doaction=True):
+    if doaction and filename is not None and os.path.exists(filename):
+        output = np.atleast_1d(np.loadtxt(filename, dtype=int)).tolist()
+    else:
+        output = []
+    return output
+
+
+########################
+### Helper Functions ###
+########################
+
+
+
+def get_skipped_expids(expid_filename, skip_expids=True):
+    return get_file_list(filename=expid_filename, doaction=skip_expids)
+
+def find_newexp(night, fileglob, known_exposures):
+    """
+    Check the path given for new exposures
+    """
+    datafiles = sorted(glob.glob(fileglob))
+    newexp = list()
+    for filepath in datafiles:
+        expid = int(os.path.basename(os.path.dirname(filepath)))
+        if (night, expid) not in known_exposures:
+            newexp.append((night, expid))
+
+    return set(newexp)
+
+def get_json_dict(reqpath):
+    req_dict = {}
+    if os.path.isfile(reqpath):
+        with open(reqpath, 'r') as req:
+            req_dict = json.load(req)
+    return req_dict
+
+def give_relevant_details(verbose_output, non_verbose_output=None, verbosely=False):
+    if verbosely:
+        print(verbose_output)
+    elif non_verbose_output is not None:
+        print(non_verbose_output)
+    else:
+        pass
+
+def define_variable_from_environment(env_name, var_descr):
+    if env_name in os.environ:
+        return os.environ[env_name]
+    else:
+        print(f'{var_descr} needs to be given explicitly or set using environment variable {env_name}')
+        exit(1)
+
+def verify_variable_with_environment(var, var_name, env_name, output_mechanism=print):
```

Comment:
`verify_variable_with_environment` and `define_variable_from_environment` feel overly specific for how infrequently they are used, which I think makes the client code less readable since it isn't clear to the naive user what these functions are doing, vs. just letting them have the slightly replicated code of e.g. checking os.environ['DESI_SPECTRO_DATA'] and letting it raise an exception if it isn't defined.

Reply:
I think it cleans up the calling scripts while making the output warning/errors more uniform, and the function names and doc strings make it clear what is being done. Otherwise, the start of all the scripts have 2-4 environment variable checks, all of which require 5-10 lines of code to nicely describe what is happening. This lets us conveniently print what we are doing so the user knows if environment variables are being used or, even more significantly, altered while not getting bogged in boilerplate code.

I am happy to discuss this further, but I don't think this is a blocking factor for this PR, so I will consider this resolved for the purposes of this PR.